src/hdmf/spec/spec.py

```diff
@@ -23,7 +23,7 @@ class DtypeHelper:
     # make sure keys are consistent between hdmf.spec.spec.DtypeHelper.primary_dtype_synonyms,
```

Comment:
Per the comment here, I tried to maintain consistency between the type maps listed. However, it is very possible that there is some side-effect from adding support for `np.bool_` that I am not aware of, so I would appreciate if someone could do a double-check for me.